src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1735,7 +1735,12 @@ private function dumpValue(mixed $value, bool $interpolate = true): string
 
                     $code = sprintf('return %s;', $code);
 
-                    return sprintf("function ()%s {\n            %s\n        }", $returnedType, $code);
+                    $attributes = '';
+                    if ($value && $this->container->hasDefinition($value) && ($class = $this->container->findDefinition($value)->getClass())) {
+                        $attributes = sprintf('#[\%s(class: %s)] ', \Closure::class, $this->dumpValue($class, $interpolate));
```

Comment:
Next to the class, would we make any use of putting the "id" of the service here (as `name` maybe?)
Not sure, I'm just wondering.

Reply:
We can add it but I don't see where this information could be displayed and if it will help users to debug.